end_to_end_tests/interface.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2020 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Interface for e2e tests."""
+
+import time
+import unittest
+import os
+import inspect
+import traceback
+from collections import Counter
```

Comment:
why not have this as `import collections`

and the counter be

`counter = collections.Counter()`

that makes it more explicit where it comes from

Reply:
Good catch - fixed.